misc/de-metas-common/de-metas-common-util/src/main/java/de/metas/common/util/FileUtil.java

```diff
@@ -64,4 +72,57 @@ public String stripIllegalCharacters(@Nullable String filename)
 
 		return sb.toString();
 	}
+
+	@NonNull
+	public Path getFilePath(@NonNull final URL url) throws MalformedURLException
+	{
+		final boolean isWindowsLocalPath = Check.isEmpty(url.getHost())
+				? url.getPath().contains(":")
+				: url.getAuthority().contains(":");
+
+		final Path path = isWindowsLocalPath
+				? parseWindowsLocalPath(url)
+				: parseFileURL(url);
+
+		if (path == null)
+		{
+			throw new RuntimeException("Couldn't parse path from:" + url);
+		}
+
+		return path;
+	}
+
+	@Nullable
+	private Path parseWindowsLocalPath(@NonNull final URL url)
+	{
+		try
+		{
+			final String normalizedPath = Stream.concat(Stream.of(url.getAuthority()), Arrays.stream(url.getPath().split("/")))
+					.filter(Check::isNotBlank)
+					.collect(Collectors.joining("/"));
+
+			return Paths.get(normalizedPath);
+		}
+		catch (final Throwable throwable)
+		{
+			return null;
+		}
+	}
+
+	@Nullable
+	private Path parseFileURL(@NonNull final URL url)
```

Comment:
what does this method do?
What's the difference between this one and parseWindowsLocalPath?
Pls make it obvious in the code (better method naming?)